packages/block-editor/src/components/link-control/index.js

```diff
@@ -171,6 +171,16 @@ function LinkControl( {
 		isEndingEditWithFocus.current = false;
 	}, [ isEditingLink ] );
 
+	/**
+	 * Update the state value internalInputValue if the url value changes
+	 * for example when clicking on another anchor
```

Comment:
```suggestion
	 * for example when clicking on another element which renders a `<LinkControl>`
```

Reply:
Might be worth documenting that this "sync state with props" is necessary because the form UI is designed not to "commit" it's changes to the controlled `value` prop unless you hit "submit". Therefore it needs to maintain an internal state value for the input which therefore must be synced with the prop.